### PR TITLE
bug fix for 'reveal' in a subdir (#903)

### LINKED ISSUE #903: `reveal` command is failed in the sub directory
https://github.com/sobolevn/git-secret/pull/751
This PR produced this issue.

- How to reproduce
```
$ git init
Initialized empty Git repository in /home/jayden/Workspace/opensource/test/.git/

$ git-secret init
git-secret: init created: '/home/jayden/Workspace/opensource/test/.gitsecret/'

$ git-secret tell test@test.com
gpg: keybox '/home/jayden/Workspace/opensource/test/.gitsecret/keys/pubring.kbx' created
gpg: /home/jayden/Workspace/opensource/test/.gitsecret/keys/trustdb.gpg: trustdb created
git-secret: done. test@test.com added as user(s) who know the secret.

$ touch aaa  

$ git-secret add aaa
git-secret: file not in .gitignore, adding: aaa
git-secret: 1 item(s) added.

$ git-secret hide
git-secret: done. 1 of 1 files are hidden.

# `reveal` worked well in the git repo root directory.
$ git-secret reveal
File '/home/jayden/Workspace/opensource/test/aaa' exists. Overwrite? (y/N) y
git-secret: done. 1 of 1 files are revealed.

$ mkdir subdir

$ cd subdir 

# `reveal` failed in the subdir.
$ git-secret reveal
git-secret: abort: cannot find file to decrypt: /home/jayden/Workspace/opensource/test/subdir/aaa.secret
```

The file path should be `/home/jayden/Workspace/opensource/test/aaa.secret` instead of `/home/jayden/Workspace/opensource/test/subdir/aaa.secret`.

### PR DESCRIPTION
**Pull request by ochanism**
resolve #903 

I've made a bug patch for `reveal` command and added a test case for covering the case.

### CODE REVIEW

File: tests/test_reveal.bats [hunk 1]
```diff
@@ -293,3 +293,25 @@ function teardown {
   # clean up
   rm -rf subdir
 }
+
+@test "run 'reveal' for all files from subdir" {
+  local password
+  password=$(test_user_password "$TEST_DEFAULT_USER")
+
+  mkdir subdir
```
Comment (sobolevn) [inline tests/test_reveal.bats:301]:
```suggestion
  mkdir -p subdir
```

File: tests/test_reveal.bats [hunk 2]
```diff
@@ -293,3 +293,25 @@ function teardown {
...
+  echo "content2" > subdir/new_filename.txt
+
+  ( # start subshell for subdir tests
+    cd subdir
+    run git secret add new_filename.txt
+    [ "$status" -eq 0 ]
+    run git secret hide
+    [ "$status" -eq 0 ]
+  
+    run git secret reveal -d "$TEST_GPG_HOMEDIR" -p "$password"
```
Comment (sobolevn) [inline tests/test_reveal.bats:311]: Let's also test that `reveal new_filename.txt` will also work!

**Review by sobolevn — COMMENTED** [review]
Thanks!

File: tests/test_reveal.bats (hunk 2 above)
Comment (ochanism) [reply tests/test_reveal.bats:311]: The test case already exists: https://github.com/ochanism/git-secret/blob/4aa2229904852e2d4b76515801a1b2e62775d3d2/tests/test_reveal.bats#L275

I copied and modified it. Do you want to merge them?

File: tests/test_reveal.bats (hunk 1 above)
Comment (ochanism) [reply tests/test_reveal.bats:301]: Got it.

File: tests/test_reveal.bats (hunk 2 above)
Comment (sobolevn) [reply tests/test_reveal.bats:311]: Ok then, let's leave it as-is.

**Review by sobolevn — APPROVED** [review]
LGTM 🎉

**Comment by joshrabinowitz** [comment]
If I'm reading correctly this merged PR is causing tests to fail?

**Comment by joshrabinowitz** [comment]
It's back to green after I reran tests. Transient error it appears.

**Pull request merged**